plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergParquetFileWriter.java

```diff
@@ -54,7 +56,8 @@ public IcebergParquetFileWriter(
             CompressionCodecName compressionCodecName,
             String trinoVersion,
             String outputPath,
-            TrinoFileSystem fileSystem)
+            TrinoFileSystem fileSystem,
+            AggregatedMemoryContext memoryContext)
```

Comment:
similar as for `OutputStreamOrcDataSink`. Create `AggregatedMemoryContext` locally in `ParquetFileWriter`.
You can then simplify calls to `createParquetFileWriter`